AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -158,11 +163,53 @@ + (void)deleteDocumentWithPartition:(NSString *)partition
   (void)completionHandler;
 }
 
-#if TARGET_OS_OSX
-- (void)dealloc {
-}
+#pragma mark - MSDataStore Implementation
+- (void)createWithPartition:(NSString *)partition
+                 documentId:(NSString *)documentId
+                   document:(id<MSSerializableDocument>)document
+               writeOptions:(MSWriteOptions *)__unused writeOptions
+          completionHandler:(MSDocumentWrapperCompletionHandler)completionHandler {
 
-#endif
+  [MSTokenExchange tokenAsync:(MSStorageIngestion *)self.ingestion
+                   partitions:@[ partition ]
+            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+              // If error getting token.
+              if (error && !tokenResponses) {
```

Comment:
Shouldn't this `&&` be `||`? Seems like if there is an error or the `tokenResponses` is `nil`, then there's an error